src/main/java/org/vaadin/addons/tatu/TabSheet.java

```diff
@@ -239,20 +249,58 @@ public void setSelected(String tab) {
         getElement().executeJs("this.setSelectedTab($0)", tab);
     }
 
+    /**
+     * Get selected tab index.
+     *
+     * @return Index of the tab, base 0 or -1 of no selection.
+     */
+    public int getSelectedIndex() {
+        return this.getElement().getProperty("selected", -1);
```

Comment:
"selected" property is not auto-synchronized. So when user changes selected tab, the value on the server side is not updated. I.e. in this form the method just returns what has been set with the corresponding setter. Check if the  property change listener as below in constructor fixes that

    public TabSheet() {
        getElement().addPropertyChangeListener("selected", "tab-changed", e -> {});
    }

Reply:
Ok, that sounds like the right way go.

Reply:
I just forgot, that there is a convenience annotation to do the same thing.

    @Synchronize(property = "selected", value = "tab-changed")
    public int getSelectedIndex() {
        return this.getElement().getProperty("selected", -1);
    }

Reply:
Implemented this, but it seems not to work all the way. Synchronizes value only once. There is something on the client-side? Needs more investigation

Reply:
Found the problem and fixed in [4b6c499](https://github.com/TatuLund/TabSheet/pull/3/commits/4b6c499b5594ef37cb75856a575ca422b407094f)